From: Michael Stapelberg <michael@stapelberg.de>
Date: Sat, 10 Aug 2013 11:06:06 +0200
Subject: =?utf-8?q?systemctl=3A_don=E2=80=99t_skip_native_units_when_enabl?=
 =?utf-8?q?ing/disabling_sysv_units?=

update-rc.d does not completely replace systemctl enable for native units;
e. g. it does not create Alias= symlinks. Thus always install/enable units
ourselves even if update-rc.d handled the corresponding SysV unit.
---
 src/systemctl/systemctl.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/systemctl/systemctl.c b/src/systemctl/systemctl.c
index 278f2f9..9a44839 100644
--- a/src/systemctl/systemctl.c
+++ b/src/systemctl/systemctl.c
@@ -5169,9 +5169,6 @@ static int enable_sysv_units(const char *verb, char **args) {
                                 break;
                 }
 
-                if (found_native)
-                        continue;
-
                 p = path_join(arg_root, SYSTEM_SYSVINIT_PATH, name);
                 if (!p)
                         return log_oom();
@@ -5181,7 +5178,7 @@ static int enable_sysv_units(const char *verb, char **args) {
                 if (!found_sysv)
                         continue;
 
-                log_info("%s is not a native service, redirecting to /usr/sbin/update-rc.d.", name);
+                log_info("Synchronizing state for %s with sysvinit using update-rc.d...", name);
 
                 if (!isempty(arg_root) && !streq(arg_root, "/")) {
                     log_error("Can not run update-rc.d when a root directory other than / is specified");
@@ -5268,6 +5265,9 @@ static int enable_sysv_units(const char *verb, char **args) {
                 } else
                         return -EPROTO;
 
+                if (found_native)
+                        continue;
+
                 /* Remove this entry, so that we don't try enabling it as native unit */
                 assert(f > 0);
                 f--;
